sdk-setup/src/rpmvalidation

```diff
@@ -0,0 +1,229 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+usage() {
+    cat <<EOF
+Run a basic quality criteria check for the given RPM file.
+
+Usage: 
+    $0 -t <target> [OPTIONS] <rpm-file>
+    
+Options:
+    -t|--target <target> The OS version to validate for. 
+                         Accepts the same argument as 'sb2 -t' does.
+    -l|--list-suites     Lists available suites
+    --suites <suite1,suite2,...> Use specified suites.
+                         If omitted, all suites are used.
+EOF
+}
+
+DEFAULT_INI_DIR=/usr/share/rpmvalidation/suites
+CONFIG_DIRS=({/usr/share,/etc}/rpmvalidation/suites)
+
+SUITES=()
+
+US=$'\x1F'
+write() ( IFS=$US; printf '%s\n' "$*"; )
+
+populate_suites() {
+    local target=$1
+    local validator_inis=()
+    SUITES=()
+    if [[ ! $target ]]; then
+        return
+    fi
+    validator_inis=($(sb2 -t "$target" ls "$DEFAULT_INI_DIR" 2>/dev/null)) || return
+    SUITES=("${validator_inis[@]%.ini}")
+}
+
+suite_value() {
+    local target=$1
+    local suite=$2
+    local key=$3
+    local type=$4
+    local ini_files=("${CONFIG_DIRS[@]/%//$suite.ini}")
+    local python=
+    read -d '' -r python <<EOF
+import sys, configparser;
+config = configparser.RawConfigParser();
+config.read(sys.argv[2:]);
+sec, val = sys.argv[1].split(".");
+if '$type' == 'boolean':
+    print(config.getboolean(sec,val));
+else:
+    print(config[sec][val]);
+EOF
+    sb2 -t "$OPT_TARGET" python3 -c "$python" "RpmValidationSuite.$key" "${ini_files[@]}"
+}
+
+suite_boolean() {
+    suite_value $1 $2 $3 boolean
+}
+
+suite_raw() {
+    suite_value $1 $2 $3 raw
+}
+ 
+list_target() {
+    local target=$1
+    local display_target=$2
+    local f=
+    for suite in "${SUITES[@]}"; do
+        local essential=$(suite_boolean "$target" "$suite" Essential)
+        if [[ "$essential" == "True" ]]; then
+            essential="Essential"
+        else
+            essential="Optional"
+        fi
+        local website=$(suite_raw "$target" "$suite" Website)
+        local name=$(suite_raw "$target" "$suite" Name)
+        write ${display_target:+"$target"} "$suite" "$essential" "$website" "$name"
+    done
+}
+
+list_all() {
+    for t in $(sdk-manage target list); do
+        populate_suites $t
+        list_target $t 1
+    done
+}
+
+run_suite() {
+    local suite=$1
+    suite_exec=$(suite_raw "$OPT_TARGET" "$suite" Exec)
+    suite_cmd=${suite_exec//%f/$OPT_RPM_FILE}
+
+    sb2 -t "$OPT_TARGET" "$suite_cmd"
+}
+
+run_all_suites() {
+    echo "Using validation suites: ${SUITES[@]}"
+    local suite=
+    for suite in "${SUITES[@]}"; do
+        run_suite $suite
+    done
+}
+
+run_selected_suites() {
+    OLD_IFS=$IFS
+    IFS=','
+    SUITES=($OPT_SUITES)
```

Comment:
Modifying OPT_SUITES after parsing the options seems like a nasty thing to do. But let's see if I can improve this.

Reply:
Now OPT_SUITES is parsed using read -a like you suggested. I also modified populate_suites so that it checks that the given suites actually exist on target.